src/main/java/duke/core/Parser.java

```diff
@@ -0,0 +1,128 @@
+                        + "didn't quite catch what you wanted to find!");
+                String searchTerm = stringCombiner(inputArray, 1, inputArray.length - 1).trim();
+                command = new FindCommand(searchTerm);
+            } else {
+
+                int index = 0;
+
+                switch (firstWord.toLowerCase()) {
+                    case "todo":
+                        if (inputArray.length <= 1) {
+                            throw new DukeException("Sorry, but I can't do anything "
+                                    + "if you don't give me the description of your todo!");
+                        }
+                        String desc = stringCombiner(inputArray, 1, inputArray.length - 1);
+                        command = AddCommand.addTodo(desc.trim());
+                        break;
+                    case "event":
+                        String[] eventSplit = textToParse.split("/at");
+                        index = indexFinder(inputArray, "/at");
+                        if (index == 0) throw new DukeException("Think you forgot the /at keyword, pardner!");
+                        if (index == 1)
+                            throw new DukeException("I'm gonna need a description for this here event!");
+                        if (eventSplit.length == 1)
+                            throw new DukeException("I'm gonna need a date or time for this!");
+                        try {
+                            command = AddCommand.addEvent(stringCombiner(inputArray, 1, index - 1).trim(),
+                                    LocalDate.parse(eventSplit[1].trim()));
+                        } catch (DateTimeParseException e) {
+                            throw new DukeException("Can't seem to make out this date over here");
+                        }
+                        break;
+                    case "deadline":
+                        String[] deadlineSplit = textToParse.split("/by");
+                        index = indexFinder(inputArray, "/by");
+                        if (index == 0) throw new DukeException("Think you forgot the /by keyword, pardner!");
+                        if (index == 1)
+                            throw new DukeException("I'm gonna need a description for this here deadline!");
+                        if (deadlineSplit.length == 1)
+                            throw new DukeException("I'm gonna need a date or time for this!");
+                        try {
+                            command = AddCommand.addDeadline(stringCombiner(inputArray, 1, index - 1).trim(),
+                                    LocalDate.parse(deadlineSplit[1].trim()));
+                        } catch (DateTimeParseException e) {
+                            throw new DukeException("Can't seem to make out this date over here");
+                        }
+                        break;
+                    default:
+                        System.out.println("Sorry, I didn't quite catch that!");
```

Comment:
Should there be an indentation for case clauses in a switch statement?

Reply:
According to the style guide, the case clauses are not indented.

![image](https://user-images.githubusercontent.com/54775106/91813034-9f329b00-ec64-11ea-9f02-6a34fc763b87.png)

Reply:
Yup, but its indented in your code haha 😅